docs/content/guides/rows/row-sorting.md

```diff
@@ -83,93 +145,502 @@ import 'handsontable/dist/handsontable.full.min.css';
+    // enable rows sorting for the entire grid
+    columnSorting: true,
+  };
+  ```
+  
+  </code-block>
+  <code-block title="Columns">
+  
+  ```js
+  const configurationOptions = {
+    columns: [
+      {
+        // enable sorting by column 1
+        columnSorting: true,
+      },
+      {
+        // disable sorting by column 2
+        columnSorting: false,
+      },
+    ],
+  };
+  ```
+  
+  </code-block>
+  <code-block title="Rows">
+  
+  ```js
+  const configurationOptions = {
+    // enable rows sorting for the entire grid
+    columnSorting: true,
+
+    // exclude rows 1 and 2 from sorting
+    afterColumnSort() {
+      yourHandsontableInstance.rowIndexMapper.moveIndexes(
+        [
+          yourHandsontableInstance.toVisualRow(0),
+          yourHandsontableInstance.toVisualRow(1)
+        ], 0);
+    },
+  };
+  ```
+  
+  </code-block>
+</code-group>
 
-Here is the list of possible options to set:
+:::
 
-* `initialConfig` (`Object`) determines the initial sort status for some column. It contains the following keys:
-  * `column` (`Number`) determines the visual index of the sorted column
-  * `sortOrder` (`String`) determines the order that the column will be sorted in - possible values: `'asc'` and `'desc'`
-* `indicator` (`Boolean`) defines whether the sorting order indicator is displayed - an arrow icon in the column header specifying the sorting order
-* `sortEmptyCells` (`Boolean`) defines whether empty cells should take part in the sorting process
-* `headerAction` (`Boolean`) defines whether clicking the header should sort the table
-* `compareFunctionFactory` (`Function`) defines the compare function factory - described in more detail in [this section](#custom-compare-functions)
+::: only-for react
 
-See the example plugin configuration below:
+<code-group>
+  <code-block title="Entire grid">
+  
+  ```jsx
+  <HotTable
+    // enable rows sorting for the entire grid
+    columnSorting={true}
+  />
+  ```
+  
+  </code-block>
+  <code-block title="Columns">
+  
+  ```jsx
+  <HotTable>
+    <HotColumn
+      // enable sorting by column 1
+      columnSorting={true}
+    />
+    <HotColumn
+      // disable sorting by column 2
+      columnSorting={false}
+    />
+  </HotTable>
+  ```
+  
+  </code-block>
+  <code-block title="Rows">
+  
+  ```jsx
+  // you need `useRef` to call Handsontable's instance methods
+  import { useRef } from 'react';
+  const hotTableComponentRef = useRef(null);
+
+  <HotTable
+    ref={hotTableComponentRef}
+
+    // enable rows sorting for the entire grid
+    columnSorting={true}
+
+    // exclude rows 1 and 2 from sorting
+    afterColumnSort={{
+      hotTableComponentRef.current.hotInstance.rowIndexMapper.moveIndexes(
+        [
+          hotTableComponentRef.current.hotInstance.toVisualRow(0),
+          hotTableComponentRef.current.hotInstance.toVisualRow(1),
+        ], 0);
+    }},
+  />
+  ```
+  
+  </code-block>
+</code-group>
+
+:::
+
+### Import the sorting module
```

Comment:
As of now, none of the examples import the `ColumnSorting` module, so the whole **Import the sorting module** section is an alternative to what's shown in the examples.

Should I rework all the examples to use the modularized approach?

Reply:
@kirszenbaum yep, you might be right. We can't just change that one page because that would ruin the consistency of the guides in this matter. I would do two things to emphasize that we have modules:

1. In the first demo on the page, I would put a commented-out line indicating that there is a way to import the module more precisely. Discuss this with the devs, please.

Now:
```
import Handsontable from 'handsontable';
import 'handsontable/dist/handsontable.full.min.css';

const container = document.querySelector('#example1');
```

Then
```
import Handsontable from 'handsontable/base';

// import all modules

// commented out below import all modules: import and register only the required modules used in the demo, such as cell type: text, date, number, time, checkbox; column sorting (or multi-column sorting depending on the demo)

```

2. The "Import (module name)" is a common section for all guides, or at least most of them. It could be placed below the core content and become a part of our template.

Reply:
That'd be quite a few lines to comment out. Here's what I believe needs to happen in each configuration:

## JS – full version

```js
import Handsontable from 'handsontable';

import 'handsontable/dist/handsontable.full.min.css';
```

## JS – through individual modules

Here, you need to import modules from `handsontable/base` instead of `handsontable`:

```js
import Handsontable from 'handsontable/base';

import 'handsontable/dist/handsontable.full.min.css';

import {
  registerPlugin,
  ColumnSorting,
  MultiColumnSorting,
} from 'handsontable/plugins';

registerPlugin(ColumnSorting);
registerPlugin(MultiColumnSorting);
```

## React – full version

Here, you need to register all modules anyway:

```jsx
import { HotTable } from '@handsontable/react';
import { registerAllModules } from 'handsontable/registry';
import 'handsontable/dist/handsontable.full.min.css';

registerAllModules();
```

## React – through individual modules

```jsx
import { HotTable } from '@handsontable/react';
import Handsontable from 'handsontable/base';

import 'handsontable/dist/handsontable.full.min.css';

import {
  registerPlugin,
  ColumnSorting,
  MultiColumnSorting,
} from 'handsontable/plugins';

registerPlugin(ColumnSorting);
registerPlugin(MultiColumnSorting);
```

How about this then:
1. We stick to one way throughout the page (and documentation): full, non-modularized version.
2. We add **Import the sorting module** at the end of the page, but we link to it from the **Enable the sorting module** section.
3. In the first demo, we add a single commented-out line: "`// to import just individual modules, see the 'Import the sorting module' section of this page`"

[0534ce9](https://github.com/handsontable/handsontable/pull/10183/commits/0534ce9b92e86272226d6b209ed92dedfe05fcd2)

Reply:
> 1. We stick to one way throughout the page (and documentation): full, non-modularized version.

Switching to modules is probably a subject for another project. We can stick with the current approach.

> 2. We add Import the sorting module at the end of the page, but we link to it from the Enable the sorting module section.

Enabling takes place after initialization, so I would skip the link and place the "importing" section somewhere at end of the article.

> In the first demo, we add a single commented-out line: "// to import just individual modules, see the 'Import the sorting module' section of this page"

Good idea, a little remix though: "// to import sorting as individual module, see the 'Import the sorting module' section on this page"